core/templates/dev/head/components/ck-editor-helpers/ck-editor-4-rte.directive.ts

```diff
@@ -95,6 +95,8 @@ angular.module('oppia').directive('ckEditor4Rte', [
           'bulletedList', 'pre', 'indent', 'outdent'];
 
         // Add external plugins.
+        // prevents ckeditor from filtering content.
+        CKEDITOR.config.allowedContent = true;
```

Comment:
From previous review:

> I am a bit confused about this. Shouldn't this already be accounted for by the extraAllowedContentRules on line 115, which makes use of the componentRule in line 61?

> Yeah, the Ckeditor behaviour is pretty unpredictable; even after adding this line, ckeditor still was removing span tags. I, too, am a bit confused in this aspect.

Please see if you can get to the bottom of this to understand what is going on. If this line isn't changing anything, we should remove it.

Reply:
@seanlip, If I remove this line, the CKEditor removes all the image ags. So  had to keep it.
@bansalnitish any idea why this is happening? Anyway I'll try to look into this.

Reply:
Hi @donosco98, thanks for looking into it. Just a general comment -- please make sure that you understand exactly what is going on in the changes you're making in the code. It's not sufficient for the code to "seem to work" (so the justification you provide is insufficient); there needs to be a throughline of reasoning and analysis that explains _why_ it's the correct change to make. Sometimes this will require deeper investigation, and it's worth doing that.

(Just wanted to point this out in order to avoid any miscommunication in the review process, since taking this approach to your PRs will probably help reduce the number of reviewer comments coming up asking "why is this happening". Thanks!)